sherpa/data.py

```diff
@@ -1233,8 +1313,7 @@ def get_staterror(self, filter=False, staterrfunc=None):
 
         """
         staterror = getattr(self, 'staterror', None)
-        filter = bool_cast(filter)
-        if filter:
+        if bool_cast(filter):
```

Comment:
Good change. Using a recerved work as a variable is rarely a good idea... this is much cleaner.

Reply:
However, it doesn't really help, since `filter` is a keyword argument to this (and many other) routines which we are not going to change, so we still have the "using a reserved work" error from tools like `pylint` for the function definition.